table/block_based/block_based_table_builder.cc

```diff
@@ -515,6 +515,9 @@ struct BlockBasedTableBuilder::Rep {
         new BlockBasedTablePropertiesCollector(
             table_options.index_type, table_options.whole_key_filtering,
             moptions.prefix_extractor != nullptr));
+    table_properties_collectors.emplace_back(
```

Comment:
Can we do this only if timestamp is enabled? Then in the TimestampTablePropertiesCollector, we can always assert timestamp_size > 0

Reply:
+1  I would expect a non-trivial performance hit as it is.

Reply:
Sure. Already fixed it